### DISCUSSION TITLE
Possibility of some features

### DISCUSSION THREAD

**Discussion by beatific-angel** [InnovateAsterisk/Browser-Phone#299] [Ideas]
I have some questions.
these are possible?

First question:
if use this phone solution, I can call with mobile phone numbers directly? this is related with dialplan of asterisk?  

second question: 
if I can receive dtmf code in web phone side when mobile phone users input dtmf code ?  

third question: 
we can change hold music and background music in webphone side? 

four question: 
if access phone url , can register automatically?

**Comment by InnovateAsterisk** [comment]
> First question:
> if use this phone solution, I can call with mobile phone numbers directly? this is related with dialplan of asterisk?

The Browser Phone is a software SIP client like any other desk phone. Think of it as a replacement for a Yealink/SNOM/Cisco/Avaya desktop phone. So the dialplan and who you can call is completely up to the setup and configuration of your Asterisk box.

> second question:
> if I can receive dtmf code in web phone side when mobile phone users input dtmf code ?

It does have a feature for this but its unused. (Remember also that Asterisk would have to forward on DTMF to the endpoint for two users to send DTMF to each other). The other question then is... if you receive a DTMF what are you going to do?

> third question:
> we can change hold music and background music in webphone side?

The music on hold is played via Asterisk, so the Browser Phone doesn't have hold music. It's and Asterisk feature. The Browser Phone has a Ring Tone, that you can change to anything.

> four question:
> if access phone url , can register automatically?

This comes down to something called provisioning. It's a big topic, and frequently asked. 

The principle is as follows: 
Asterisk and SIP only really have two Authentication options, Digest and IP Auth. That is, either with a username and password, or with the IP address (that you connect from). Most of the time you use Digest Authentication, that is a system of handshakes that take place with the SIP signal of REGISTER. Once this is complete, the session may be initiated with an INVITE. It's very simple, and yes a username and password is used as part of this process - even though the actual password is not sent via SIP, it needs to be available to the  code to respond to the auth challenge. Most people see this as somewhat insecure - but it depends on your use-case. 

Ultimately this project somehow requires the sip password variable or DB field to be populated with the correct password. How you get this populated tho, is up to you. A typical approach would be to have the index.html converted into a index.php page (or index .aspx), and have the server save the sip details in a database. Then when the user logs in via your own login process (or LDAP or oAuth etc), the real SIP details are loaded into the javascript variable at runtime. This way you are not sending around the real SIP details, but they are being used "behind the scenes".

There is much more to this process, but this is the basic approach.

**Comment by beatific-angel** [comment]
in web phone side, it need only receiving dtmf code
where is the feature?
I want to use the feature in webphone

**Comment by beatific-angel** [comment]
web phone 's location should be same with freepbx, asterisk?

**Reply by bbakirtas** [reply]
i try in asterisk , fusionpbx and web hosting works on all
